lib/puppet/property/list.rb

```diff
@@ -45,7 +45,8 @@ def delimiter
 
       def retrieve
         #ok, some 'convention' if the list property is named groups, provider should implement a groups method
-        if provider and tmp = provider.send(name) and tmp != :absent
+        tmp = provider.send(name) if provider
+        if tmp && tmp != :absent
```

Comment:
Actually we want the version I have here in both because otherwise `tmp` could be `nil`, which would pass the `!= :absent` check, but was previously being guarded against. I'll change keyvalue.rb to match this file's convention.